framework/src/bundle/CoreBundleContext.cpp

```diff
@@ -169,20 +169,32 @@ void CoreBundleContext::Uninit1()
   listeners.Clear();
   resolver.Clear();
 
+  // Do not hold the bundleThreads lock while calling
+  // BundleThread::Quit() because this will join the bundle
+  // thread which may need to acquire the bundleThreads lock
+  // itself.
+  //
+  // At this point, all bundles are stopped and all
+  // bundle context instances invalidated. So no new bundle
+  // threads can be created and it is sufficient to get
+  // the current list once and not check for new bundle
+  // threads again.
+  std::list<std::shared_ptr<BundleThread>> threads;
+  bundleThreads.Lock(), std::swap(threads, bundleThreads.value);
```

Comment:
I think that this leads to a similar, yet different, situation with ```bundleThreads.value```. Once the ```bundleThreads``` locks is released, ```bundleThreads.value``` could be modified by a ```BundleThread``` thread before ```Quit()``` is called.

What if ```BundleThread::Join()``` was called for all ```BundleThread``` objects first? It appears that this would ensure that no thread will ever modify ```bundleThreads.value```.